### ISSUE TITLE
Error with Quantizing MNIST with tf.lite

### ISSUE THREAD

**Issue by dhingratul** [tensorflow/tensorflow#20653]
Please go to Stack Overflow for help and support:

https://stackoverflow.com/questions/tagged/tensorflow

If you open a GitHub issue, here is our policy:

1. It must be a bug, a feature request, or a significant problem with documentation (for small docs fixes please send a PR instead).
2. The form below must be filled out.
3. It shouldn't be a TensorBoard issue. Those go [here](https://github.com/tensorflow/tensorboard/issues).

**Here's why we have that policy**: TensorFlow developers respond to issues. We want to focus on work that benefits the whole community, e.g., fixing bugs and adding features. Support only helps individuals. GitHub also notifies thousands of people when issues are filed. We want them to see you communicating an interesting problem, rather than being redirected to Stack Overflow.

------------------------

### System information
- **Have I written custom code (as opposed to using a stock example script provided in TensorFlow)**: yes
- **OS Platform and Distribution (e.g., Linux Ubuntu 16.04)**: ubuntu 16.04
- **TensorFlow installed from (source or binary)**: bazel
- **TensorFlow version (use command below)**:  commit bcf7e315b4031b3c355af12ca2a4961bcd25c248
- **Python version**:  2.7
- **Bazel version (if compiling from source)**: Build label: 0.14.1
- **GCC/Compiler version (if compiling from source)**:
- **CUDA/cuDNN version**:
- **GPU model and memory**: 1080ti
- **Exact command to reproduce**:

`bazel build tensorflow/contrib/lite/toco:toco && \
  ./bazel-bin/tensorflow/contrib/lite/toco/toco \
  --input_file=comple_path/frozen_model.pb \
  --output_file=tflite_model.tflite \
  --input_format=TENSORFLOW_GRAPHDEF --output_format=TFLITE \
  --inference_type=QUANTIZED_UINT8 \
  --input_shape="1,28, 28,1" \
  --input_array="Inputs" \
  --output_array="fc2/Relu"  `

You can collect some of this information using our environment capture script:

https://github.com/tensorflow/tensorflow/tree/master/tools/tf_env_collect.sh

You can obtain the TensorFlow version with

python -c "import tensorflow as tf; print(tf.GIT_VERSION, tf.VERSION)"

### Describe the problem
Describe the problem clearly here. Be sure to convey here why it's a bug in TensorFlow or a feature request.

`2018-07-09 14:00:04.623621: I tensorflow/contrib/lite/toco/import_tensorflow.cc:1090] Converting unsupported operation: Assign
2018-07-09 14:00:04.623726: I tensorflow/contrib/lite/toco/import_tensorflow.cc:1090] Converting unsupported operation: Assign
2018-07-09 14:00:04.623752: I tensorflow/contrib/lite/toco/import_tensorflow.cc:1090] Converting unsupported operation: AssignAdd
2018-07-09 14:00:04.623829: I tensorflow/contrib/lite/toco/import_tensorflow.cc:1090] Converting unsupported operation: AssignSub
2018-07-09 14:00:04.623846: I tensorflow/contrib/lite/toco/import_tensorflow.cc:1090] Converting unsupported operation: AssignAdd
2018-07-09 14:00:04.623879: I tensorflow/contrib/lite/toco/import_tensorflow.cc:1090] Converting unsupported operation: AssignSub
2018-07-09 14:00:04.623914: I tensorflow/contrib/lite/toco/import_tensorflow.cc:1090] Converting unsupported operation: AssignSub
2018-07-09 14:00:04.623930: I tensorflow/contrib/lite/toco/import_tensorflow.cc:1090] Converting unsupported operation: AssignAdd
2018-07-09 14:00:04.623961: I tensorflow/contrib/lite/toco/import_tensorflow.cc:1090] Converting unsupported operation: AssignSub
2018-07-09 14:00:04.624008: I tensorflow/contrib/lite/toco/import_tensorflow.cc:1090] Converting unsupported operation: Assign
2018-07-09 14:00:04.624030: I tensorflow/contrib/lite/toco/import_tensorflow.cc:1090] Converting unsupported operation: Assign
2018-07-09 14:00:04.624098: I tensorflow/contrib/lite/toco/import_tensorflow.cc:1090] Converting unsupported operation: AssignSub
2018-07-09 14:00:04.624114: I tensorflow/contrib/lite/toco/import_tensorflow.cc:1090] Converting unsupported operation: AssignAdd
2018-07-09 14:00:04.624146: I tensorflow/contrib/lite/toco/import_tensorflow.cc:1090] Converting unsupported operation: AssignSub
2018-07-09 14:00:04.624178: I tensorflow/contrib/lite/toco/import_tensorflow.cc:1090] Converting unsupported operation: AssignSub
2018-07-09 14:00:04.624193: I tensorflow/contrib/lite/toco/import_tensorflow.cc:1090] Converting unsupported operation: AssignAdd
2018-07-09 14:00:04.624224: I tensorflow/contrib/lite/toco/import_tensorflow.cc:1090] Converting unsupported operation: AssignSub
2018-07-09 14:00:04.624270: I tensorflow/contrib/lite/toco/import_tensorflow.cc:1090] Converting unsupported operation: Assign
2018-07-09 14:00:04.624292: I tensorflow/contrib/lite/toco/import_tensorflow.cc:1090] Converting unsupported operation: Assign
2018-07-09 14:00:04.624355: I tensorflow/contrib/lite/toco/import_tensorflow.cc:1090] Converting unsupported operation: AssignSub
2018-07-09 14:00:04.624371: I tensorflow/contrib/lite/toco/import_tensorflow.cc:1090] Converting unsupported operation: AssignAdd
2018-07-09 14:00:04.624403: I tensorflow/contrib/lite/toco/import_tensorflow.cc:1090] Converting unsupported operation: AssignSub
2018-07-09 14:00:04.624436: I tensorflow/contrib/lite/toco/import_tensorflow.cc:1090] Converting unsupported operation: AssignSub
2018-07-09 14:00:04.624451: I tensorflow/contrib/lite/toco/import_tensorflow.cc:1090] Converting unsupported operation: AssignAdd
2018-07-09 14:00:04.624482: I tensorflow/contrib/lite/toco/import_tensorflow.cc:1090] Converting unsupported operation: AssignSub
2018-07-09 14:00:04.624522: I tensorflow/contrib/lite/toco/import_tensorflow.cc:1090] Converting unsupported operation: Assign
2018-07-09 14:00:04.624547: I tensorflow/contrib/lite/toco/import_tensorflow.cc:1090] Converting unsupported operation: Assign
2018-07-09 14:00:04.624606: I tensorflow/contrib/lite/toco/import_tensorflow.cc:1090] Converting unsupported operation: AssignSub
2018-07-09 14:00:04.624621: I tensorflow/contrib/lite/toco/import_tensorflow.cc:1090] Converting unsupported operation: AssignAdd
2018-07-09 14:00:04.624652: I tensorflow/contrib/lite/toco/import_tensorflow.cc:1090] Converting unsupported operation: AssignSub
2018-07-09 14:00:04.624685: I tensorflow/contrib/lite/toco/import_tensorflow.cc:1090] Converting unsupported operation: AssignSub
2018-07-09 14:00:04.624700: I tensorflow/contrib/lite/toco/import_tensorflow.cc:1090] Converting unsupported operation: AssignAdd
2018-07-09 14:00:04.624732: I tensorflow/contrib/lite/toco/import_tensorflow.cc:1090] Converting unsupported operation: AssignSub
2018-07-09 14:00:04.627783: I tensorflow/contrib/lite/toco/graph_transformations/graph_transformations.cc:39] Before Removing unused ops: 229 operators, 362 arrays (0 quantized)
2018-07-09 14:00:04.630029: I tensorflow/contrib/lite/toco/graph_transformations/graph_transformations.cc:39] After Removing unused ops pass 1: 173 operators, 282 arrays (0 quantized)
2018-07-09 14:00:04.632293: I tensorflow/contrib/lite/toco/graph_transformations/graph_transformations.cc:39] Before general graph transformations: 173 operators, 282 arrays (0 quantized)
2018-07-09 14:00:04.672712: I tensorflow/contrib/lite/toco/graph_transformations/graph_transformations.cc:39] After general graph transformations pass 1: 107 operators, 201 arrays (1 quantized)
2018-07-09 14:00:04.674373: I tensorflow/contrib/lite/toco/graph_transformations/graph_transformations.cc:39] After general graph transformations pass 2: 107 operators, 201 arrays (1 quantized)
2018-07-09 14:00:04.676019: I tensorflow/contrib/lite/toco/graph_transformations/graph_transformations.cc:39] Before pre-quantization graph transformations: 107 operators, 201 arrays (1 quantized)
2018-07-09 14:00:04.676780: F tensorflow/contrib/lite/toco/tooling_util.cc:1612] Array conv1/act_quant/AssignMinEma/conv1/act_quant/min/Pow, which is an input to the Sub operator producing the output array conv1/act_quant/AssignMinEma/conv1/act_quant/min/sub_2, is lacking min/max data, which is necessary for quantization. Either target a non-quantized output format, or change the input graph to contain min/max information, or pass --default_ranges_min= and --default_ranges_max= if you do not care about the accuracy of results.
Aborted (core dumped)
`

I tried it with default_ranges_min, max and it throws the following error
`
2018-07-09 13:56:44.667379: E tensorflow/core/util/command_line_flags.cc:124] Couldn't interpret value  for flag default_ranges_min.
2018-07-09 13:56:44.667451: E tensorflow/core/util/command_line_flags.cc:124] Couldn't interpret value  for flag default_ranges_max.
`

![screenshot from 2018-07-09 14-03-56](https://user-images.githubusercontent.com/4759327/42475903-ae68adc6-83bb-11e8-9317-fe565127f784.png)

### Source code / logs
Include any logs or source code that would be helpful to diagnose the problem. If including tracebacks, please include the full traceback. Large logs and files should be attached. Try to provide a reproducible test case that is the bare minimum necessary to generate the problem.

**Comment by raghuraman-k** [comment]
It looks like you are trying to convert a training graph, please note that TOCO only supports converting eval graphs.

**Comment by dhingratul** [comment]
Can you expand a bit on it, I did follow the workflow on the documentation page, and did call the create_eval function.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @raghuraman-k: It has been 14 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by suharshs** [comment]
You should have one graph that is a training graph built with create_training_graph. Then you should create a separate graph that is the corresponding eval graph (just the forward pass, no gradients) that you should create with create_eval_graph.

Then you should freeze the graph with the freeze_graph tool and provide to tflite_convert.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @raghuraman-k: It has been 14 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by wincle** [comment]
@suharshs Could you please tell me what's the correlation with these two graphs?
How could I use my training graph to generate the eval graph, in other words, what's the point of creating training graph , how should I use it?
Thx!
